projects/plugins/jetpack/modules/sharedaddy/sharing.php

```diff
@@ -122,26 +124,26 @@ public function process_requests() {
 
 	/**
 	 * Register Sharing settings menu page.
+	 *
+	 * We only need to register it on self-hosted sites, in offline mode.
 	 */
 	public function subscription_menu() {
-		if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
-			$active = Jetpack::get_active_modules();
-			if (
-				! in_array( 'publicize', $active, true )
-				&& ! current_user_can( 'manage_options' )
-			) {
-				return;
-			}
+		// Hide the menu on the wpcom platform.
+		if ( ( new Host() )->is_wpcom_platform() ) {
+			return;
 		}
 
-		add_submenu_page(
-			'options-general.php',
-			__( 'Sharing Settings', 'jetpack' ),
-			__( 'Sharing', 'jetpack' ),
-			'publish_posts',
-			'sharing',
-			array( $this, 'wrapper_admin_page' )
-		);
+		// Register it only on self-hosted sites, in offline mode.
+		if ( Jetpack::is_module_active( 'sharedaddy' ) && ( new Status() )->is_offline_mode() ) {
```

Comment:
I don't think we need the active module check here since that file is only called when the module is active.

Reply:
I made changes to register the menu in the offline mode on ALL sites as the link is still pointed to the Sharing page when people add `define( 'JETPACK_DEV_DEBUG', true );` to `wp-config.php` on the wpcom platform (the atomic sites).

Reply:
I don't think we should aim to support Offline Mode on the Atomic platform. The sites are always publicly accessible there, and the WordPress.com connection is necessary to manage the site via WordPress.com.

Reply:
> The sites are always publicly accessible there

Sorry, not sure if it's relevant but does the private site setting matter here? It's broken a few things in the past.

![Screenshot 2024-02-22 at 12-56-05 General Settings ‹ My WordPress Site — WordPress com](https://github.com/Automattic/jetpack/assets/811776/ff1024ae-2c5f-49e1-924b-e199243cecd3)

Reply:
Yep. I assume it won't break the connection as well.

Reply:
https://github.com/Automattic/wpcomsh/blob/revert-1688-update/nav-redesign-enabled-site-visibility-override-remove/private-site/private-site.php#L33

Can ignore private sites from this, disables a few things including sharing.

Reply:
That indeed wouldn't break the connection, and that's indeed something we do not need to worry about here, since the Sharing module isn't enabled on private sites, and thus the code in `projects/plugins/jetpack/modules/sharedaddy/sharing.php` doesn't run.